src/main/java/edu/usf/sas/pal/muser/manager/MuserMusicDataAnalysisManager.java

```diff
@@ -45,15 +46,42 @@ public MuserMusicDataAnalysisManager() throws FirebaseFileNotInitializedExceptio
     public void processData() {
         // create csv file and add the header
         csvFileWriter.createHeader(MusicAnalysisModel.CSV_HEADER);
-        if (programOptions.getUserId() == null) {
+
+        long startDateMillis = programOptions.getStartDate();
+        long endDateMillis = programOptions.getEndDate();
+
+        if (startDateMillis > 0 && endDateMillis > 0 && programOptions.getUserId() == null) {
+            // Only date range is provided, make a filtered query for all users using date range
+            if (startDateMillis < endDateMillis) {
```

Comment:
So, I think that moving this validation to main is better. Since we already validate in main that startDate and endDate should be greater than 0, I think this extra check here is redundant. We can add the check for `startDate < endDate` along with that as well. 

But here in the manager class, we need to check for cases where dates were provided, or just userId, or none. Thus, I believe we still need some sort of flag to check if dates were provided. So, I was thinking of adding a flag inside main. We can validate the dates inside main and make this flag true and use it in manager for our dates provided check. 

Does this way sound good?

Reply:
I would just put the "are dates provided" code into a utility method and call it both places.